app/models/post.rb

```diff
@@ -1,2 +1,15 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def self.update_post_counter(user)
+    count = Post.where("author_id = #{user.id}").count
+    user.posts_counter = count
+    user.save
+  end
```

Comment:
- [ ] [Optional] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private